### DISCUSSION TITLE
GKE Autopilot cluster gets created with default service account

### DISCUSSION THREAD

**Discussion by kkalchuri-solutelabs** [pulumi/pulumi#15902] [General]
Hi Team,
I am trying to create a GKE private **Autopilot** cluster via Pulumi typescript. I am creating a custom service account with limited IAM permission and want to attach the same to autopilot cluster instead of using default service account.

I tried multiple code changes but my Autopilot cluster is getting created with `default` service account only.  Using [this ](https://www.pulumi.com/registry/packages/gcp/api-docs/container/cluster/#gcp-container-cluster) Pulumi package for autopilot cluster creation.
I came across terraform issue discussion where many users are facing similar default account issue.
https://github.com/hashicorp/terraform-provider-google/issues/9505

I tried the listed solution to use  `clusterAutoscaling` but Pulumi says it conflicts with the `enable_autopilot`, same behavior with node pool, you can't change much in Autopilot cluster.

Why Pulumi or terraform is failing to attach custom service account to autopilot cluster?
Any suggestions I can try?
     
```
import * as pulumi from "@pulumi/pulumi";
import * as gcp from "@pulumi/gcp";

const sa = new gcp.serviceaccount.Account("sa", {
    createIgnoreAlreadyExists: true,
    accountId: "test",
});
const project = new gcp.projects.IAMBinding("project", {
    project: "<Project_ID>",
    role: "roles/cloudsql.admin",
    members: [sa.email.apply(email => `serviceAccount:${email}`)],
});

const primary = new gcp.container.Cluster("primary", {
    name: "cluster-test",
    network: "<VPC_ID>",
    subnetwork: "<Private_Subnet_ID>",
    location: "us-east4",
    enableAutopilot: true,
    deletionProtection: false,
    masterAuthorizedNetworksConfig: {
        cidrBlocks: [{
            cidrBlock: "<IP>",
        }],
    },
    privateClusterConfig: {
        enablePrivateEndpoint: false,
        enablePrivateNodes: true,
    },
    nodeConfig: {
        serviceAccount: sa.email
    }
});
```

**Comment by solomonshorser** [comment]
I'm having this problem too, using pulumi gcp package 7.29.0

I opened an issue on this here: https://github.com/pulumi/pulumi-gcp/issues/2124

**Reply by hayesgm** [reply]
For those still facing this issue, it looks like this solution works to get the correct service account, cribbed from https://github.com/hashicorp/terraform-provider-google/issues/9505#issuecomment-1340074019:

```ts
new gcp.container.Cluster(name,
  {
    enableAutopilot: true,
    // ...
    nodeConfig: {
      serviceAccount: gkeNodeServiceAccount.email
    },
    clusterAutoscaling: {
      autoProvisioningDefaults: {
        serviceAccount: gkeNodeServiceAccount.email
      }
    }
  }
);
```

Pulumi Versions

```
NAME                VERSION
@pulumi/gcp         7.29.0
@pulumi/kubernetes  3.30.2
@pulumi/pulumi      3.122.0
```
